src/privacy-policy.js

```diff
@@ -0,0 +1,63 @@
+let utils = require('./utils')
+
+class PrivacyPolicy {
+    addLists (lists) {
+        this.tosdrList = lists.tosdr
+        this.polisisList = lists.polisis
+
+        this.addDuplicateRulesForParents(this.tosdrList)
+        this.addDuplicateRulesForParents(this.polisisList)
+    }
+
+    getScoreForUrl (url) {
+        let hostname = utils.extractHostFromURL(url)
+        let domain = utils.getDomain(url)
+        let parent = utils.findParent(hostname.split('.'))
+
+        // prefer tosdr data because that's generated by humans
+        let privacyData = this.tosdrList[parent] ||
+            this.tosdrList[domain] ||
+            this.tosdrList[hostname]
```

Comment:
**Q:** I've copied the approach from [here](https://github.com/duckduckgo/privacy-grade/blob/9376fa575eb68f5323707160f6dc4c71d369fcdd/grade-scripts/post-process/grade.js#L383-L406), basically check the parent entity first, then the actual hostname.

I'm not sure about this. Wouldn't it make sense to pick the most specific privacy policy data we've got, and then go general if we don't have privacy policy data?

As an example, if we've got privacy policy data for Youtube and Google, we should prefer YT.